manageiq-operator/pkg/helpers/miq-components/httpd.go

```diff
@@ -534,3 +546,32 @@ func tlsSecretName(cr *miqv1alpha1.ManageIQ) string {
 
 	return secretName
 }
+
+func fetchIntrospectionUrl(providerUrl string) (string, error) {
+	customTransport := http.DefaultTransport.(*http.Transport).Clone()
+	customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	client := &http.Client{Transport: customTransport}
+	errMsg := fmt.Sprintf("failed to get the OIDCOAuthIntrospectionURL from %s", providerUrl)
+
+	resp, err := client.Get(providerUrl)
+	if err != nil {
+		return "", fmt.Errorf("%s - %s", errMsg, err)
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("%s - StatusCode: %d", errMsg, resp.StatusCode)
+	}
+
+	var result map[string]interface{}
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return "", fmt.Errorf("%s - %s", errMsg, err)
+	}
+
+	if result["token_introspection_endpoint"] == nil {
+		return "", fmt.Errorf("%s - token_introspection_endpoint is missing", errMsg)
```

Comment:
Maybe "token_introspection_endpoint is missing **from response**"?
I don't think I would understand what this was saying as is.